controllers/apply/building-connections/processor.js

```diff
@@ -26,7 +26,7 @@ module.exports = async function processor(formModel, formData) {
     const stepsWithValues = formModel.getStepsWithValues(formData);
 
     const dataToStore = formatDataForStorage(stepsWithValues);
-    const record = await applicationService.storeApplication(formModel.shortCode, dataToStore);
+    const record = await applicationService.storeApplication(formModel.id, formModel.shortCode, dataToStore);
```

Comment:
(I've pre-emptively added this field manually to the existing dev/prod databases so we can merge this without issues)